### ISSUE TITLE
Have an article explaining when (or how to decide) to use DBI vs dplyr

### ISSUE THREAD

**Issue by bborgesr** [rstudio/db.rstudio.com#41]
This is something that comes up whenever I talk about databases, and it would be good to have "official" guidelines that we can direct people to. Since DBI and dplyr overlap a lot in their database functionality, users (especially newbies) are often at a loss on the topic of which one to choose.

In the olden times, when dplyr only supported SELECT-power statements, the rule was easy. Now, not so much. However, that may still be a good rule of thumb as non-SELECT statements in dplyr have not been time-tested yet.

Whatever the official recommendation is, we should have one, so that the answer to "Which package to use?" is not completely the opinion of whomever happens to be answering it.

cc @hadley

**Comment by hadley** [comment]
While thinking through whether or not dplyr should have `append_to()` as an analog to `copy_to()` I realised one absolute principle of dplyr is that it never mutates data in place - it always creates a copy with the modifications. That means that `append_to()` is explicitly out of scope, and in general if you want to modify existing data you should be using DBI, not dplyr.

**Comment by hadley** [comment]
Why would you be writing to a database table as part of an analysis?

**Comment by edgararuiz** [comment]
Yes, the example will start with a database that already has data we want to work with, as it is in most real  cases. So, I see two scenarios we need to address:

- The user has a small data set in R that they want to join to the data in the database 

- A query that results form `dplyr` code takes to long to re-run every time, so a temp table is needed

**Comment by nwstephens** [comment]
Maybe instead of setting strict guidelines, we just do a comparison of methods between the two and leave it to the reader to decide. For example, I'm fairly certain that some users who want to do a select star will use dbgetquery even though there are many other ways to print a table.

**Comment by edgararuiz** [comment]
That's why I was thinking of an example.  To me, this is a softer way to suggest a guideline that the user can take or leave.
